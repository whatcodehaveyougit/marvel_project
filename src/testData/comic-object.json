{
  "id": 47176,
  "digitalId": 30661,
  "title": "FREE COMIC BOOK DAY 2013 1 (2013) #1",
  "issueNumber": 1,
  "variantDescription": "",
  "description": "",
  "modified": "2019-06-25T17:31:51-0400",
  "isbn": "",
  "upc": "75960607889900111",
  "diamondCode": "",
  "ean": "",
  "issn": "",
  "format": "Comic",
  "pageCount": 32,
  "textObjects": [
    {
      "type": "issue_solicit_text",
      "language": "en-us",
      "text": "It's out action featuring the newest stars to be of Disney XD: Avengers and the Hulk and the Agents of S.M.A.S.H.! Featuring art pulled directly from the animated series, this sneak peek at the two newest animated series from the geniuses at Marvel Animation promises to make this the best yet!"
    }
  ],
  "resourceURI": "http://gateway.marvel.com/v1/public/comics/47176",
  "urls": [
    {
      "type": "detail",
      "url": "http://marvel.com/comics/issue/47176/free_comic_book_day_2013_1_2013_1?utm_campaign=apiRef&utm_source=022caac83153a4aff24a965cdf020ba7"
    },
    {
      "type": "purchase",
      "url": "http://comicstore.marvel.com/FREE-COMIC-BOOK-DAY-2013-1-1/digital-comic/30661?utm_campaign=apiRef&utm_source=022caac83153a4aff24a965cdf020ba7"
    },
    {
      "type": "reader",
      "url": "http://marvel.com/digitalcomics/view.htm?iid=30661&utm_campaign=apiRef&utm_source=022caac83153a4aff24a965cdf020ba7"
    },
    {
      "type": "inAppLink",
      "url": "https://applink.marvel.com/issue/30661?utm_campaign=apiRef&utm_source=022caac83153a4aff24a965cdf020ba7"
    }
  ],
  "series": {
    "resourceURI": "http://gateway.marvel.com/v1/public/series/17765",
    "name": "FREE COMIC BOOK DAY 2013 1 (2013)"
  },
  "variants": [],
  "collections": [],
  "collectedIssues": [],
  "dates": [
    {
      "type": "onsaleDate",
      "date": "2013-03-20T00:00:00-0400"
    },
    {
      "type": "focDate",
      "date": "2013-03-04T00:00:00-0500"
    },
    {
      "type": "unlimitedDate",
      "date": "2013-11-04T00:00:00-0500"
    },
    {
      "type": "digitalPurchaseDate",
      "date": "2013-07-09T00:00:00-0400"
    }
  ],
  "prices": [
    {
      "type": "printPrice",
      "price": 0.2
    },
    {
      "type": "digitalPurchasePrice",
      "price": 0
    }
  ],
  "thumbnail": {
    "path": "http://i.annihil.us/u/prod/marvel/i/mg/9/50/57ed5bc9040e3",
    "extension": "jpg"
  },
  "images": [
    {
      "path": "http://i.annihil.us/u/prod/marvel/i/mg/9/50/57ed5bc9040e3",
      "extension": "jpg"
    }
  ],
  "creators": {
    "available": 3,
    "collectionURI": "http://gateway.marvel.com/v1/public/comics/47176/creators",
    "items": [
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/creators/13586",
        "name": "Joe Caramagna",
        "role": "writer"
      },
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/creators/5251",
        "name": "Vc Joe Caramagna",
        "role": "letterer"
      },
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/creators/11879",
        "name": "Stephen Wacker",
        "role": "editor"
      }
    ],
    "returned": 3
  },
  "characters": {
    "available": 5,
    "collectionURI": "http://gateway.marvel.com/v1/public/comics/47176/characters",
    "items": [
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/characters/1017100",
        "name": "A-Bomb (HAS)"
      },
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/characters/1017098",
        "name": "Hulk (HAS)"
      },
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/characters/1017099",
        "name": "Red Hulk (HAS)"
      },
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/characters/1017111",
        "name": "She-Hulk (HAS)"
      },
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/characters/1017101",
        "name": "Skaar (HAS)"
      }
    ],
    "returned": 5
  },
  "stories": {
    "available": 2,
    "collectionURI": "http://gateway.marvel.com/v1/public/comics/47176/stories",
    "items": [
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/stories/105929",
        "name": "cover from Free Comic Book Day 2013 (Avengers/Hulk) (2013) #1",
        "type": "cover"
      },
      {
        "resourceURI": "http://gateway.marvel.com/v1/public/stories/105930",
        "name": "story from Free Comic Book Day 2013 (Avengers/Hulk) (2013) #1",
        "type": "interiorStory"
      }
    ],
    "returned": 2
  },
  "events": {
    "available": 0,
    "collectionURI": "http://gateway.marvel.com/v1/public/comics/47176/events",
    "items": [],
    "returned": 0
  }
}
