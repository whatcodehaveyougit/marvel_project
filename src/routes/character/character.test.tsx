import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
// import Character from "./character";
import * as characterData from "./testData/character-object.json";
import * as comicData from "./testData/comic-object.json";
import * as ReactRouterDom from "react-router-dom";
// import { useSelector } from 'react-redux';
import * as reactRedux from "react-redux";
import { jest, describe, it, expect } from "@jest/globals";
import Character from "./character";

jest.mock("react-redux", () => ({
  // ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(() => jest.fn()),
  Provider: ({ children }: any) => children,
}));

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as typeof ReactRouterDom),
  useParams: jest.fn().mockReturnValue({ characterid: 1011334 }),
}));

// jest.mock("../../components/spinner/spinner.component.tsx", () =>
//   jest.fn(() => "Spinner")
// );

console.log(characterData);

const mockStore = configureStore([]);

describe("Character", () => {
  // const useSelectorMock = jest.spyOn(reactRedux, 'useSelector')
  // beforeEach(() => {
  //   const useParamsMock = jest.fn().mockReturnValue({ characterid: 1011334 });
  //   ReactRouterDom.useParams = useParamsMock;
  //   ReactRouterDom.useParams.mockReturnValue({ characterid: 1011334 });
  // });
  it("Spinner is rendered when there is no comic book data but the data is loading", () => {
    const initialState = {
      characters: {
        data: null,
        isLoading: true,
        error: null,
      },
      character: {
        data: characterData,
        isLoading: true,
        error: null,
      },
      characterComics: {
        data: [comicData],
        isLoading: true,
        error: null,
      },
    };
    const store = mockStore(initialState);
    jest
      .spyOn(reactRedux, "useSelector")
      .mockImplementation((callback) => callback(store.getState()));
    console.log(store.getState());
    const view = render(
      <Provider store={store}>
        <Character />
      </Provider>
    ).asFragment();
    expect(view).toMatchSnapshot();
  });
  // it("Accordion of comics is rendered when data has loaded in", () => {
  //   const initialState1 = {
  //     characters: [],
  //     isLoading: false,
  //     error: null,
  //     character: {
  //       comics: [comicData],
  //       data: { name: "THe hulk" },
  //       isLoading: false,
  //       error: null,
  //     },
  //   };
  //   const store = mockStore(initialState1);
  //   jest
  //     .spyOn(reactRedux, "useSelector")
  //     .mockImplementation((callback) => callback(store.getState()));
  //   const view = render(
  //     <Provider store={store}>
  //       <Character />
  //     </Provider>
  //   );
  //   expect(view).toMatchSnapshot();
  // });
});
