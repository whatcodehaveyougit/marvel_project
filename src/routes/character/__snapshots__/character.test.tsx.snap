// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Spinner is rendered when there is no comic book data but the data is loading 1`] = `
<DocumentFragment>
  <div
    class="character-page bg-gray-100"
  >
    <div
      class="thumbnail-image bg-cover bg-center"
      style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg);"
    >
      <div
        class="character-information p-8"
      >
        <div
          class="character-page-headings"
        >
          <h3
            class="text-3xl text-center"
          >
            3-D Man Test
          </h3>
          <h5
            class="text-lg text-center"
          >
            List of comics for this character:
          </h5>
        </div>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
